### PR TITLE
BCDA-1243 Feature: save-public-key CLI command

### PR DESCRIPTION
**Pull request by dhgreene**
### Fixes [BCDA-1243](https://jira.cms.gov/browse/BCDA-1243)
Building on BCDA-1322, this PR adds a CLI command for saving an ACO's public key. 

(Please review PR #268 first)

### Proposed changes:
New CLI command
Screen Shot 2019-05-22 at 9 18 01 AM

### Change Details
Call `aco.SavePublicKey()` in `bcda/models/models.go`

### Security Implications
This CLI command saves the contents of a specified file into the database.  Security controls include host and network controls that limit CLI access to team members, and file verifications in `aco.SavePublicKey()`.

### Acceptance Validation
save-public-key

### Feedback Requested
- Testing is done in the style of other tests in `bcda/bcdacli/cli_test.go`.  Is now the time to refactor them?
- Is there a more secure (or otherwise better) way of opening files?

### CODE REVIEW

File: bcda/bcdacli/cli.go [hunk 1]
```diff
@@ -170,6 +172,57 @@ func setUpApp() *cli.App {
...
+					return errors.New("key-file is required")
+				}
+
+				aco, err := auth.GetACOByCMSID(acoCMSID)
+				if err != nil {
+					fmt.Fprintf(app.Writer, "Unable to find ACO %s: %s\n", acoCMSID, err.Error())
+					return err
+				}
+
+				/* #nosec -- Potential file inclusion via variable */
```
Comment (knollfear) [inline bcda/bcdacli/cli.go]: This line is unnecessary now that you have the filepath.Clean command in the line below.  I ran it locally to confirm.

File: bcda/bcdacli/cli_test.go [hunk 2]
```diff
@@ -238,6 +238,67 @@ func (s *CLITestSuite) TestCreateToken() {
...
+	assert.Equal("cms-id is required", err.Error())
+	assert.Contains(buf.String(), "")
+
+	// Unspecified File
+	args = []string{"bcda", "save-public-key", "--cms-id", "A9901", }
+	err = s.testApp.Run(args)
+	assert.Equal("key-file is required", err.Error())
+	assert.Contains(buf.String(), "")
+
+	// Non-existant ACO
```
Comment (knollfear) [inline bcda/bcdacli/cli_test.go]:
```suggestion
	// Non-existent ACO
```

File: bcda/bcdacli/cli.go (hunk 1 above)
Comment (dhgreene) [reply bcda/bcdacli/cli.go]: OK.

**Comment by dhgreene** [comment]
> I think the tests should be broken up into smaller individual tests instead of one big one. Even if we haven't fixed the old ones we should stop digging as it just makes more work in the future and will be easier to fix now.

@knollfear I took this seriously and performed the refactoring as an exercise locally this morning.  I am now of the opinion that it's better code as a single test, especially because three of the cases being tested require creation of a new ACO.  They might as well share the same ACO.

I could possibly be persuaded otherwise, but have been advised by @DeirdreHolub  to get this code committed soonest.  Would one of you care to approve (again) as soon as CI tests run again?  Thanks!

**Pull request merged**